Source/WebCore/page/Quirks.cpp

```diff
@@ -1332,22 +1323,23 @@ bool Quirks::needsConfigurableIndexedPropertiesQuirk() const
 // Canvas fingerprinting (rdar://107564162)
 String Quirks::advancedPrivacyProtectionSubstituteDataURLForScriptWithFeatures(const String& lastDrawnText, int canvasWidth, int canvasHeight) const
 {
-    if (!needsQuirks() || !m_document->settings().canvasFingerprintingQuirkEnabled() || !m_document->noiseInjectionHashSalt())
+    Ref document = *m_document;
```

Comment:
This is wrong. `m_document` may be null. The previous code was fine because it called `needsQuirks()` before using `m_document` and `needsQuirks()` nulls checks `m_document`.